tests/tracer/test_provider.py

```diff
@@ -0,0 +1,218 @@
+import threading
+from typing import Dict
+from typing import Optional
+from typing import Union
+
+from ddtrace.context import Context
+from ddtrace.provider import BaseContextProvider
+from ddtrace.provider import DatadogContextMixin
+from ddtrace.span import Span
+
+
+class TestContextProvider(BaseContextProvider, DatadogContextMixin):
```

Comment:
It's weird to write a test context that you test rather than testing the real one? (or am I reading too fast?)